host/platform/Linux/capture/XCB/src/xcb.c

```diff
@@ -293,30 +308,65 @@ static int pointerThread(void * unused)
     uint32_t size;
     if (!this->getPointerBufferFn(&data, &size))
     {
-      DEBUG_WARN("failed to get a pointer buffer");
+      DEBUG_WARN("Failed to get a pointer buffer");
       continue;
     }
 
     xcb_xfixes_get_cursor_image_reply_t * curReply;
     curReply = xcb_xfixes_get_cursor_image_reply(this->xcb, this->curC, NULL);
-    data = xcb_xfixes_get_cursor_image_cursor_image(curReply);
-
-    if (!data)
+    if (curReply == NULL)
     {
-      DEBUG_WARN("xcb_xfixes_get_cursor_image failed");
+      DEBUG_WARN("Failed to get cursor reply");
       continue;
     }
 
-    this->mouseVisible = pointer.visible;
-    this->mouseHotX    = pointer.hx;
-    this->mouseHotY    = pointer.hy;
+    if(curReply->xhot != this->mouseHotX || curReply->yhot != this->mouseHotY)
+    {
+      pointer.shapeUpdate = true;
+      this->mouseHotX = curReply->xhot;
+      this->mouseHotY = curReply->yhot;
 
-    pointer.positionUpdate = true;
-    pointer.visible        = this->mouseVisible;
-    pointer.x              = this->mouseX - pointer.hx;
-    pointer.y              = this->mouseY - pointer.hy;
+      uint32_t * src = xcb_xfixes_get_cursor_image_cursor_image(curReply);
+      uint32_t * dst = (uint32_t *) (data);
+      memcpy(dst, src, curReply->width * curReply->height * sizeof(uint32_t));
+    }
+    else
+    {
+      pointer.shapeUpdate = false;
+    }
+
+    if(curReply->x != this->mouseX || curReply->y != this->mouseY)
+    {
+      pointer.positionUpdate = true;
+      this->mouseX = curReply->x;
+      this->mouseY = curReply->y;
+    }
+    else
+    {
+      pointer.positionUpdate = false;
+    }
+
+    if(!pointer.positionUpdate && !pointer.shapeUpdate)
+    {
+      free(curReply);
+      nsleep(1000);
```

Comment:
any reason this is `nsleep` and not `usleep(1)`? Also this is a very short sleep, i'd suggest this is too fast.

Reply:
I did not have `#include <uinstd.h>` before so usleep was not included. I believe I originally intended it to be `usleep(1000)`/`sleep(1)` and scale back from there if the responsiveness wasn't optimal. I'll make that change.

The code did work without the sleep but obviously had much higher CPU usage than necessary, hence the addition. Ideally, it'd likely be better to sleep until a change but at the moment I wasn't sure how to do that in xcb given the poor documentation online.